### DISCUSSION TITLE
Support 'Metadata-only' load for EXIF/IPTC manipulation cases

### DISCUSSION THREAD

**Discussion by Webreaper** [SixLabors/ImageSharp#1784] [Ideas]
Somewhat related to #1371 but this time for Metadata operations - it would be great if there was a way to load a metadata-only representation of an image without decoding the image data itself. 

The primary use-case would be for operations where you want to load the image, add/remove/update EXIF/IPTC metadata, and write the metadata back to the original image file (or a copy). By not loading the entire image into the stream, or decoding it, these operations could be made significantly faster (particularly for very large image files). 

Currently I use [an out-of-process instance of ExifTool ](https://github.com/Webreaper/Damselfly/blob/master/Damselfly.Core/Services/MetaDataService.cs#L232) to update image metadata, but if ImageSharp could do it all with approximately the same performance, I'd drop that immediately. 

Bonus feature would be to implement the ability to load the image data too, for the use case:

```
var metaOnlyImage = LoadMetaDataOnly( path );

if( metaOnlyImage.MetaData.Iptc.Keywords.Contains ( "Interesting" ) )
{
     metaOnlyImage.LoadImageData();
     // Do something interesting here
}
```
Although I suspect that it may just be easier to reload a new instance of the image from scratch at that point, so it might be unnecessary.

**Comment by antonfirsov** [comment]
@Webreaper loading is supported via [`Image.Identify`](https://docs.sixlabors.com/api/ImageSharp/SixLabors.ImageSharp.Image.html#SixLabors_ImageSharp_Image_Identify_System_String_). It returns an `IImageInfo` with [`Metadata`](https://docs.sixlabors.com/api/ImageSharp/SixLabors.ImageSharp.IImageInfo.html#SixLabors_ImageSharp_IImageInfo_Metadata) property filled in.

We don't support metadata-only updating though.

**Reply by Webreaper** [reply]
I hadn't! But I'll bear that in mind. 

The killer would be metadata-only updating, but it's possible I may be able to replace the MetadataExtractor with Image.Identify etc. Will give it a try.

**Reply by JimBobSquarePants** [reply]
We're still [missing XMP metadata](https://github.com/SixLabors/ImageSharp/issues/858) support. We're currently capturing it internally in TIFF but have no public API as yet.
